Clang "unused function EnableClmul" warning on 32-bit systems

Compiling Bitcoin Core, the recent master branch (https://github.com/bitcoin/bitcoin/commit/57982f419e36d0023c83af2dd0d683ca3160dc2a):

```
$ clang-12 --version
Ubuntu clang version 12.0.0-3ubuntu1~20.04.4
Target: armv7l-unknown-linux-gnueabihf
Thread model: posix
InstalledDir: /usr/bin
$ make
...
  CXX      minisketch/src/libminisketch_a-minisketch.o
minisketch/src/minisketch.cpp:66:20: warning: unused function 'EnableClmul' [-Wunused-function]
static inline bool EnableClmul()
                   ^
1 warning generated.
...
```

--- a/src/minisketch.cpp
+++ b/src/minisketch.cpp
@@ -63,9 +63,9 @@ enum class FieldImpl {
 #endif
 };
 
+#ifdef HAVE_CLMUL
 static inline bool EnableClmul()
 {
-#ifdef HAVE_CLMUL
 #ifdef _MSC_VER
     int regs[4];
     __cpuid(regs, 1);
@@ -74,10 +74,8 @@ static inline bool EnableClmul()
     uint32_t eax, ebx, ecx, edx;
     return (__get_cpuid(1, &eax, &ebx, &ecx, &edx) && (ecx & 0x2));
 #endif
-#else
-    return false;
-#endif
 }
+#endif
 
 Sketch* Construct(int bits, int impl)
 {

